src/modules/auth/auth.controller.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable require-jsdoc */
+import Joi from 'joi';
+import validation from '../../helpers/validation';
+import Auth from '../../helpers/Auth';
+import response from '../../helpers/response';
+import users from '../../services/userServices';
+
+class authController {
+  static async signin(req, res) {
+    const schema = validation.validateSignin();
```

Comment:
Validation is supposed to be a middleware inside `./src/middleware` folder. Since I was already working on this PR, you can use this method below

Create file `./src/middlewares/ValidateAuth.js` and input code below
```import Joi from '@hapi/joi';
import joiValidator from '../helpers/joiValidator';
import responseHandler from '../helpers/responseHandler';

/**
 *   Class to validate user login and signup forms submission
 */
class ValidateAuth {
  /**
   * validate email and password
   * @param {Object} req
   * @param {Object} res
   * @param {Function} next
   * @returns {Object} error
   */
  static validateSigninFormData(req, res, next) {
    const password = Joi.string().required();
    const email = Joi.string().email().required();

    const createSignUpSchema = Joi.object().keys({
      password,
      email,
    });

    const errors = joiValidator(req.body, createSignUpSchema);
    if (!errors) {
      return next();
    }
    return responseHandler.validationError(res, errors);
  }
}

export default ValidateAuth;
```
create an helper function in `./src/helper/joiValidator.js` that validates the above, put below code
```import Joi from '@hapi/joi';

/**
* @param {Object} data
* @param {Object} schema
* @param {Function} next
* @returns {Function} next middleware
*/
const joiValidator = (data, schema) => {
  let message;
  const validationOptions = {
    allowUnknown: true, // allow unknown keys that will be ignored
    stripUnknown: true, // remove unknown keys from the validated data
    abortEarly: false // validate all inputs befor flagging error
  };
  Joi.validate(data, schema, validationOptions, (err) => {
    if (err) {
      message = err.details.map((i) => i.message.replace(/['"]/g, ''));
    }
  });
  return message;
};

export default joiValidator;```

all you need to do now is to include the `ValidateAuth` middleware in your route.